agent/engine/service_connect/manager_linux.go

```diff
@@ -15,17 +15,84 @@ package serviceconnect
 
 import (
 	"fmt"
+	"os"
+	"path/filepath"
 
 	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
 	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
 	dockercontainer "github.com/docker/docker/api/types/container"
 )
 
+const (
+	defaultRelayPathContainer  = "/var/run/ecs/relay/"
+	defaultRelayPathHost       = "/var/run/ecs/instance/relay/"
+	defaultRelayFileName       = "envoy_xds.sock"
+	defaultRelayENV            = "APPMESH_XDS_ENDPOINT"
+	defaultStatusPathContainer = "/var/run/ecs/"
+	defaultStatusPathHostRoot  = "/var/run/ecs/"
+	defaultStatusFileName      = "appnet_admin.sock"
+	defaultStatusENV           = "APPNET_AGENT_UDS_PATH"
+)
+
 type manager struct {
+	relayPathContainer  string
+	relayPathHost       string
+	relayFileName       string
+	relayENV            string
+	statusPathContainer string
+	statusPathHostRoot  string
+	statusFileName      string
+	statusENV           string
 }
 
 func NewManager() Manager {
-	return &manager{}
+	return &manager{
+		relayPathContainer:  defaultRelayPathContainer,
+		relayPathHost:       defaultRelayPathHost,
+		relayFileName:       defaultRelayFileName,
+		relayENV:            defaultRelayENV,
+		statusPathContainer: defaultStatusPathContainer,
+		statusPathHostRoot:  defaultStatusPathHostRoot,
+		statusFileName:      defaultStatusFileName,
+		statusENV:           defaultStatusENV,
+	}
+}
+
+func IsServiceConnectEnabledTask(task *apitask.Task) bool {
+	return task.IsServiceConnectEnabled()
+}
+
+func (m *manager) initializeAgentContainer(task *apitask.Task, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+	return m.initServiceConnectDirectoryMounts(task.GetID(), container, hostConfig)
+}
+
+func getBindMountMapping(hostDir, containerDir string) string {
+	return hostDir + ":" + containerDir
```

Comment:
these mappings should probably read only since it's at the directory level. The actual UDS file will have write permissions, but the directory itself should not.

```golang
	return hostDir + ":" + containerDir + ":ro"
```

Reply:
Will SC Agent and Relay need directory write permission to create the socket files?

Reply:
yes they will...

Reply:
Can ECS agent create the UDS files instead and the relay will just bind to them?

Reply:
of that's not an option, I think at least I'd prefer to change the directories to `/var/run/ecs/service_connect` to limit blast radius in case of a  bad actor trying to write stuff.

Reply:
Nope the UDS file and thus the inode resulting from its creation is effectively the thing that is listened on or conencted to. I can add a SC suffix for sure.